### PR TITLE
feat(j-s): Send service update emails to prosecutors

### PR DESCRIPTION
**Pull request by oddsson**
# Send service update emails to prosecutors

[Asana](https://app.asana.com/0/1199153462262248/1208607083085532/f)

## What

When a subpoena service update is made (either successful or failed), the court users get a notification about that. Prosecutors need to get these notification also. This PR implements this.

We're introducing a new way of sending routes in emails in this PR. 

**Old way**
Send hardcoded routes in email notifications. 

**New way**
Send route`/beinir/[caseId]` in all emails. The component behind this route gets the case by case id and redirects the user based on the case state and other factors.

This solves a few issues

- The routing rules are moved into a single place, rather than being implemented where the notification is sent.
- Eliminates a issue where we send a route to f.x. an open indictment overview and the recipient don't open it until the indictment is completed. The user should be routed to the completed indictment overview page but that doesn't happen today, since the route is hardcoded in the notification. 

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Enhanced email notification system to support multiple recipients.
  - Introduced a new `RouteHandler` component for improved routing based on user roles and case statuses.

- **Improvements**
  - Updated email formatting for links to ensure accurate navigation.
  - Added a loading container style for better user experience during data processing.

- **Documentation**
  - New file created for `RouteHandler`, facilitating its integration within the application.

### CODE REVIEW

**Review by gudjong — APPROVED** [review]

File: apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+const getRoute = (caseToOpen?: Case, user?: User): string => {
+  if (!caseToOpen || !user) {
+    return '/'
+  }
+
+  const userType: UserType | null = isProsecutionUser(user)
+    ? 'prosecution'
+    : isDistrictCourtUser(user)
+    ? 'districtCourt'
+    : null
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx:61]: I disagree, I like ternary operators

File: apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import {
+  isCompletedCase,
+  isDistrictCourtUser,
+  isIndictmentCase,
+  isProsecutionUser,
+} from '@island.is/judicial-system/types'
+import {
+  FormContext,
+  UserContext,
+} from '@island.is/judicial-system-web/src/components'
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx:20]: `@island.is/judicial-system-web/src/components` is a shared index.

File: apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        )
+      }
+    },
+    [caseToOpen, getCase, router],
+  )
+
+  useEffect(() => {
+    handleGetCase(router.query.id?.toString())
+    router.push(getRoute(caseToOpen, user))
+  }, [caseToOpen, handleGetCase, queryCase, router, user])
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx]: Thanks, I agree!

File: apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    : null
+  const caseStatus = getCaseStatus(caseToOpen.state)
+
+  const route =
+    caseToOpen.type &&
+    userType &&
+    routes[caseToOpen.type]?.[userType]?.[caseStatus]
+
+  return route ? `${route}/${caseToOpen.id}` : '/'
+}
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx:70]: No but thanks

File: apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+      if (!caseToOpen) {
+        getCase(
+          caseId,
+          (caseData) => setCaseToOpen(caseData),
+          () => router.push('/'),
+        )
+      }
+    },
+    [caseToOpen, getCase, router],
+  )
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Shared/RouteHandler/RouteHandler.tsx:98]: No but thanks

**Pull request merged**